srcpkgs/shadow/patches/xstrdup.patch

```diff
@@ -1,9 +1,10 @@
---- a/libmisc/xmalloc.c	2008-08-30 21:55:44.000000000 -0500
-+++ b/libmisc/xmalloc.c.new	2008-08-30 21:55:36.000000000 -0500
-@@ -61,5 +61,6 @@
- 
- char *xstrdup (const char *str)
+--- a/lib/alloc.h
++++ b/lib/alloc.h
+@@ -108,6 +108,7 @@ reallocarrayf(void *p, size_t nmemb, size_t size)
+ inline char *
+ xstrdup(const char *str)
  {
-+	if(str == NULL) return NULL;
- 	return strcpy (xmalloc (strlen (str) + 1), str);
++	if (str == NULL) return NULL;
+ 	return strcpy(XMALLOC(strlen(str) + 1, char), str);
```

Comment:
These xfoo() functions are designed to exit/abort on errors reported by the corresponding foo() function.

Thus, xstrdup() is strdup(3) plus a check of an output NULL from strdup(3) to error out.  The precise reason to have these functions is so that they can *never* return NULL.

Adding null checks to a NULL input is tangential to the purpose of these functions, and defeats their purpose.  Now the function can return NULL, which needs to be checked by the caller.

So, if you really want to check for programmer errors, you'd need to exit on a NULL input, not just pass it through.

But you probably shouldn't do that at all.  strdup(3) doesn't handle NULL input, and I don't see a reason why xstrdup() should.

Reply:
This patch goes back a long time (introduced in Void in 2012, shadow 4.1.5), so many things may have changed. If you say it's not needed, I tend to trust you. Maybe @q66 can comment as to why Chimera still keeps it?

Reply:
Hmmm, I'd prefer to know why it originated, before confirming that it should be removed, but most likely.  Let's hear @q66 .  :)